src/components/fab/EphemeralAudioPlayerFab.tsx

```diff
@@ -0,0 +1,267 @@
+
+  const containerStyle = React.useMemo(() => [{ width: remainingSpace }], [
+    remainingSpace,
+  ]);
+
+  const currentPlayingAssetTitle = React.useMemo(() => {
+    if (!!currentlyPlayingAsset && typeof currentlyPlayingAsset === 'object') {
+      // @ts-ignore
+      const { name } = currentlyPlayingAsset;
+      return `${name}`;
+    }
+    return '';
+  }, [currentlyPlayingAsset]);
+
+  const currentPlayingAssetDetailsColor = isDarkMode
+    ? colors.orangeLight
+    : colors.orangeLight;
+
+  const currentPlayingAssetDetailsStyle = React.useMemo(
+    () => ({
+      color: currentPlayingAssetDetailsColor,
+    }),
+    [currentPlayingAssetDetailsColor]
+  );
+  const handleOnPressToggleAudio = React.useCallback(async () => {
+    if (isPlayingAsset && currentSound) {
+      // @ts-ignore
+      const shouldPause = () => currentSound.pause();
+      // @ts-ignore
+      const shouldPlay = () => currentSound.play();
+      return isPlayingAssetPaused ? shouldPlay() : shouldPause();
+    }
+    const randomAsset = pickRandomAsset();
+    // @ts-ignore
+    return playAsset(randomAsset);
+  }, [
+    currentSound,
+    isPlayingAsset,
+    isPlayingAssetPaused,
+    playAsset,
+    pickRandomAsset,
+  ]);
+  const handleOnPressSkip = React.useCallback(async () => {
+    // @ts-ignore
+    return playAsset(pickNextAsset());
+  }, [playAsset, pickNextAsset]);
+  return (
+    <Animated.View pointerEvents="box-none" {...props} style={containerStyle}>
+      {/* Overflow Container */}
+      <Animated.View
+        pointerEvents="box-none"
+        style={[
+          styles.absolute,
+          styles.overflowContainer,
+          styles.noOverflow,
+          { width: remainingSpace - paddingRight },
+        ]}
+      >
+        {/* Animated Background */}
+        <Animated.View
+          pointerEvents={open ? 'auto' : 'none'}
+          style={[
+            styles.absolute,
+            styles.background,
+            styles.noOverflow,
+            styles.details,
+            {
+              backgroundColor: colors.white,
+              borderRadius: FloatingActionButtonSize * 0.5,
+              transform: [
+                {
+                  translateX: Animated.multiply(
+                    Animated.subtract(1, progress),
+                    remainingSpace - (FloatingActionButtonSize + paddingRight)
+                  ),
+                },
+              ],
+              width: remainingSpace - paddingRight,
+            },
+          ]}
+        >
+          <ButtonPressAnimation
+            disabled={disabled || !currentlyPlayingAsset}
+            onPress={handleAudioTitlePress}
+            opacity={isDarkMode && disabled ? 0.6 : 1}
+            scaleTo={0.95}
+            {...props}
+          >
+            <Bold
+              children={
+                currentlyPlayingAsset ? currentPlayingAssetTitle : 'My Playlist'
+              }
+              numberOfLines={1}
+              style={{
```

Comment:
We prefer to use styled-components over inline styles

Reply:
We've commented out this component because it needs a little UX design, so to fix this I'm going to temporarily add a `.env` flag `EXPERIMENTAL_AUDIO_PLAYER` which when set to `true` will render this component. This way I can make the changes but I won't be interfering with existing expected behaviour.